drivers/i2c/i2c_smartbond.c

```diff
@@ -425,13 +437,13 @@ static int i2c_smartbond_transfer_cb(const struct device *dev, struct i2c_msg *m
 	data->cb = cb;
 	data->userdata = userdata;
 
+	i2c_smartbond_pm_policy_state_lock_get(dev);
+
 	ret = i2c_smartbond_prep_transfer(dev, msgs, num_msgs, addr);
 	if (ret != 0) {
 		return ret;
```

Comment:
Since policy is already locked it should be released before returning with error code by calling `i2c_smartbond_pm_policy_state_lock_put`

Reply:
Fixed